src/log/log.c

```diff
@@ -2832,3 +2783,68 @@ __wt_log_flush(WT_SESSION_IMPL *session, uint32_t flags)
         WT_RET(__wt_log_force_sync(session, &lsn));
     return (0);
 }
+
+/*
+ * __log_fsync_dir --
+ *     Perform fsync of log->log_dir_fh. Requires log->log_sync_lock to be held by the caller.
+ */
+int
```

Comment:
This should be static (and moved further up in the file before the callers). (I suggest after `log_fs_write`)